src/shared/credentials/userCredentialsUtils.ts

```diff
@@ -72,6 +72,18 @@ export class UserCredentialsUtils {
             || path.join(SystemUtilities.getHomeDirectory(), '.aws', 'config')
     }
 
+    /**
+     * @description Determines if credentials directory exists
+     * If it doesn't, creates credentials directory
+     * at directory from this.getCredentialsFilename()
+     */
+    public static async generateCredentialDirectoryIfNonexistent(): Promise<void> {
+        const filepath = path.dirname(this.getCredentialsFilename())
+        if (!await fileExists(filepath)) {
+            await mkdir(filepath)
```

Comment:
To be safe, use `await mkdir(filepath, { recursive: true })`

Reply:
✔️